### ISSUE TITLE
[BUG] Inngest not working with hono

### ISSUE THREAD

**Issue by BernardoSM** [inngest/inngest-js#556]
**Describe the bug**
3 days ago was merged hono framework for inngest that you can see [here](https://github.com/inngest/inngest-js/commit/24f1e7de92d609e0a9ee9df1e3dfa0d2f7c8f682).

I tried to run the [example](https://github.com/inngest/inngest-js/tree/main/examples/framework-hono) and works pretty well, but when I run [inngest dev server](https://www.inngest.com/docs/local-development#inngest-dev-server) I'm receiving some errors.

**To Reproduce**

1. Follow [Getting started](https://github.com/inngest/inngest-js/tree/main/examples/framework-hono#getting-started), run app locally and open `http://localhost:3000/api/inngest` to check if inngest is working
2. Run `npx inngest-cli@latest dev` in terminal

**Expected behavior**
Not show any error on terminal that is running hono (cloudflare worker) with inngest.
Not show two apps on inngest dev server

**Code snippets / Logs / Screenshots**

<details><summary>Hono terminal errors:</summary>

```bash
[wrangler:inf] GET /api/inngest 500 Internal Server Error (2ms)
✘ [ERROR] workerd/server/server.c++:3340: error: Uncaught exception: kj/async-io-unix.c++:569: disconnected: ::read(fd, buffer, maxBytes): Connection reset by peer

  stack:
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@55c6383
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@55c43e5
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@405a071
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40778a0
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40bfeb1
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40835e0
  workerd/server/server.c++:3340: error: Uncaught exception:
  kj/async-io-unix.c++:186: disconnected: ::write(fd, buffer, size):
  Connection reset by peer
  stack:
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@55c508f
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@55c5ba1
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@4062df9
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40632f6
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@5583fff
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40be61b
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40bf54b
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40bff82
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40835e0
  workerd/server/server.c++:3135: error: kj::getCaughtExceptionAsKj()
  = kj/async-io-unix.c++:186: disconnected: ::write(fd, buffer,
  size): Connection reset by peer
  stack:
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@55c508f
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@55c5ba1
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@4062df9
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40632f6
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@5583fff
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40bd5ea
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40c0cfd
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@40c0fc4
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@348c0e0
  /home/bernardosm/blindpaylabs2/inngest-hono/node_modules/@cloudflare/workerd-linux-64/bin/workerd@2e0b1a0


[wrangler:inf] GET /x/inngest 404 Not Found (1ms)
[wrangler:inf] GET /.netlify/functions/inngest 404 Not Found (1ms)
[wrangler:inf] GET /.redwood/functions/inngest 404 Not Found (1ms)
[wrangler:inf] PUT /api/inngest 200 OK (4ms)
[wrangler:inf] PUT /api/inngest 200 OK (4ms)
[wrangler:inf] PUT /api/inngest 500 Internal Server Error (2ms)
✘ [ERROR] [Inngest] error - TypeError: Failed calling `url` from serve handler when starting to handle request; Invalid URL string.

      at url
  (file:///home/bernardosm/blindpaylabs2/inngest-hono/node_modules/src/hono.ts:38:20)
      at fn
  (file:///home/bernardosm/blindpaylabs2/inngest-hono/node_modules/src/components/InngestCommHandler.ts:529:16)
      at async InngestCommHandler.handleAction
  (file:///home/bernardosm/blindpaylabs2/inngest-hono/node_modules/src/components/InngestCommHandler.ts:700:19)
      at async ServerTiming.wrap
  (file:///home/bernardosm/blindpaylabs2/inngest-hono/node_modules/src/helpers/ServerTiming.ts:83:15)
      at async ServerTiming.wrap
  (file:///home/bernardosm/blindpaylabs2/inngest-hono/node_modules/src/helpers/ServerTiming.ts:83:15)
      at async jsonError
  (file:///home/bernardosm/blindpaylabs2/inngest-hono/node_modules/wrangler/templates/middleware/middleware-miniflare3-json-error.ts:22:10)
      at async drainBody
  (file:///home/bernardosm/blindpaylabs2/inngest-hono/node_modules/wrangler/templates/middleware/middleware-ensure-req-body-drained.ts:5:10)
```

</details>

<details><summary>Inngest dev server trying to connect at the same app twice:</summary>

![image](https://github.com/inngest/inngest-js/assets/36281298/f7680f5a-0a3b-41c7-ac9d-7629628eb5b3)

</details>

**System info (please complete the following information):**

- OS: Ubuntu 22.04.1 LTS (Running on windows + WSL2)
- npm package Version: 10.1.0
- Framework: Hono
- Platform: Cloudflare Workers
- Node: v20.9.0

**Comment by BernardoSM** [comment]
I found a workaround for it, I'm running `npx inngest-cli@latest dev --no-poll --no-discover` and syncing app manually.

**Comment by arjunyel** [comment]
@BernardoSM This is working for me with Cloudflare Workers and Hono

```typescript
app.on(["GET", "PUT", "POST"], "/api/inngest", (c) => {
  const handler = serve({
    client: inngest,
    functions,
    signingKey: c.env.INNGEST_SIGNING_KEY,
  });
  return handler(c);
});
```

**Comment by arjunyel** [comment]
Ok I am not sure what the issue is but the example is using port 3000. If you serve the Cloudflare worker at its regular port of 8787 it works with `npx inngest-cli dev -u http://localhost:8787/api/inngest`

**Comment by jpwilliams** [comment]
Thanks for the reports, folks!

I've added a fix for some of these issues over at #571 and you're right, @BernardoSM: inngest/inngest#1274 would definitely have been causing the double-app issue.

I'd love it if folks could test out that the fix at #571 works. You can check it out using `npm install inngest@pr-571`.

For the large `workerd` errors, e.g.:
```
workerd/server/server.c++:3340: error: Uncaught exception: kj/async-io-unix.c++:569: disconnected: ::read(fd, buffer, maxBytes): Connection reset by peer
```
I'm still seeing those even when everything working fine, and also it's kind of sporadic and not on every poll. 🤔 It may be an issue with the way the Dev Server is closing requests (especially the discovery requests which are intended to be very low cost), so definitely something we can address.

**Comment by jpwilliams** [comment]
I can't reproduce these worker errors any longer; it may well have been an issue that's resolved in a later version of workerd.

cloudflare/workerd#1401 is still open and it looks exceedingly transient.

Please feel free to re-open this issue if anyone sees this again and we can contribute to the discussion/solution there.
